### DISCUSSION TITLE
Using Aviary to only simulate and not optimize a specified mission

### DISCUSSION THREAD

**Discussion by superlil27** [OpenMDAO/Aviary#499] [Q&A] [answered]
I am looking to use Aviary to run sweeps of missions to explore my aircraft's performance. I tried setting constant speed segments to make mission integration faster but this was not working with IPOPT (unconverged trajectory is shown below for the aircraft_for_bench_FwFm_out vehicle). 
![image](https://github.com/user-attachments/assets/bf79fd9e-9c2b-4ace-a90e-84cf09a9b995)

I also tried removing the range and target range constraint and set durations for each segment but this didn't work either.

Is there a way to run Aviary to just simulate a mission once without it trying to minimize fuel burn or anything else?

**Comment by jkirk5** [comment, marked as answer]
By default Aviary uses the collocation method to solve the trajectory - this method is powerful, but always requires optimization, as that is how the method also ensure that the chosen trajectory is feasible. In your case, since you don't mind if a trajectory isn't _optimal_, just _feasible_, a shooting method is what you want to use. We have a mostly complete implementation of the SGM method in Aviary, but it has some limitations (search the docs for "SGM" for specifics).

Try following along with [this example](https://github.com/OpenMDAO/Aviary/blob/main/aviary/examples/level2_shooting_traj.py), which uses the Level 2 Python interface

**Reply by superlil27** [reply]
Thank you! I'll go through that example.

**Reply by superlil27** [reply]
While running the example I didn't quite get an output.
![image](https://github.com/user-attachments/assets/ae7b07ac-0642-40d9-a111-82a329cbce73)

It also created a bunch of output folders with just the inputs in them.
![image](https://github.com/user-attachments/assets/5dc33aff-4b2e-4370-8ada-6aad225d9d1d)

This might be what you referred to as "mostly complete implementation of the SGM method". Just wanted to share it here in case others had a similar issue.
